### PR TITLE
Create a Guidelines page for check-in workflow

### PR DESCRIPTION
**Pull request by sashadev-sky**
References #4692 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] PR is descriptively titled 📑
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by rexagod** [comment]
We can ask the long timers and active ones to document out stuff that can be potentially very useful for the new ones. These guidelines will be different than what you find in a README file. They will include workarounds, examples of different instances where the contributors got stuck and felt the need to inform others about this so that they don't waste their time on the same. All that information which seems a little informal to be included in the README files but also very helpful to be left out. 

Something like `guide.publiclab.org` maybe,  where contributors can post content like the above so that the contributors don't get stuck, or out of frustration, leave. We can also add the ability for the contributors to get feedback from other members to help them out, much like a forum I suppose (sorry if I'm being a little ambitious here :smile: ). Currently there isn't any apt place for the newer members to clear out these doubts -- they can't post them on `publiclab.org` since that won't be appropriate; the Gitter chat is very much perceptible to questions and doubts getting lost in the humungous amount of messages that occur daily, leaving no other option. 

`guide.publiclab.org` can be a platform where contributors can ask each other questions and obtain feedback for the same. We can keep a strict set of rules such as a question has not been asked before, is not already present in the issues, etc. This was just a thought, and if you find this interesting and/or want to suggest some modifications do let me know and we can elaborate on this.

Thanks!

**Comment by rexagod** [comment]
We can ask the long timers and active ones to document out stuff that can be potentially very useful for the new ones. These guidelines will be different than what you find in a README file. They will include workarounds, examples of different instances where the contributors got stuck and felt the need to inform others about this so that they don't waste their time on the same. All that information which seems a little informal to be included in the README files but also very helpful to be left out. 

Something like `guide.publiclab.org` maybe,  where contributors can post content like the above so that the contributors don't get stuck, or out of frustration, leave. We can also add the ability for the contributors to get feedback from other members to help them out, much like a forum I suppose (sorry if I'm being a little ambitious here :smile: ). Currently there isn't any apt place for the newer members to clear out these doubts -- they can't post them on `publiclab.org` since that won't be appropriate; the Gitter chat is very much perceptible to questions and doubts getting lost in the humungous amount of messages that occur daily, leaving no other option. 

`guide.publiclab.org` can be a platform where contributors can ask each other questions and obtain feedback for the same. We can keep a strict set of rules such as a question has not been asked before, is not already present in the issues, etc. This was just a thought, and if you find this interesting and/or want to suggest some modifications do let me know and we can elaborate on this.

Thanks!

**Comment by sashadev-sky** [comment]
@rexagod I love this idea! I have been thinking about that as well. I have a file I started making when I first joined where I aggregated different links from the site that I found useful for starting out. And I would also paste in another file random bits of information people have told me. I also think a page where we can write repo setup issues we have encountered and solutions for them would be so helpful. 

We can make wikis on these maybe? For a social platform that can be part of community toolbox planning list?

**Comment by sashadev-sky** [comment]
@gauravano please see my changes @rexagod in my 'add anything else' point with example topics I will add a bullet about what you are describing above

**Comment by gauravano** [comment]
Hey @rexagod nice points, although you really dived into the project :sweat_smile:. It's true that we need someplace because many new contributors don't open an issue for their problem. Having a forum would allow us to mention our knowledge of issues we have already faced, so thinkable as our community is growing. Maybe we can think of integrating with an existing platform (in the future, if not now).Or, we can also try to expand tech-discussions to publiclab.org, drop links in our readme for posting problems there ,etc.

**Comment by gauravano** [comment]
@sashadev-sky, actually let's add this one - 

> Adding a link to check-in for reference, to help the user a little 

This will allow the new contributors to raise their hands for check-in as they would feel comfortable with a reference in hand. Thanks!

File: doc/CHECKINS.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  - Have any questions? 
+
+  - Troubleshot any issues that the rest of the community can learn from?
+
+- Include sections from the last post that make sense to carry over - if you need additional guidance reference past check-ins.
+
+## Thanks! 
+
+Thank you for helping us maintain our community! 🎉 🌍 🌎 🌏
```
[the remark this answers is not in the document]
Comment (sashadev-sky) [reply doc/CHECKINS.md]: Done!

**Comment by sashadev-sky** [comment]
@gauravano I am a little unclear about your other revision, is it possible you could request a revision on top of my latest commit with the change you want? You want the link updated or no?

File: doc/CHECKINS.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Community Check-ins
+
+One of our community values is openness and diversity in thought. For this reason, we have started weekly check-ins run by our contributors (you!). The check-ins are currently opened as issues on the plots2 repository [issues](https://github.com/publiclab/plots2/issues?utf8=%E2%9C%93&q=) page
```
[the remark this answers is not in the document]
Comment (gauravano) [reply doc/CHECKINS.md]: Yes @sashadev-sky , you need to merge this suggestion.

**Comment by gauravano** [comment]
Hi @sashadev-sky, I made some changes and I think it's complete now. If you feel any change is required, feel free to add/suggest. Thanks!

**Comment by sashadev-sky** [comment]
@gauravano ok great, changes will be merged tomorrow and i'll look over it as a whole again :)

**Comment by sashadev-sky** [comment]
@gauravano updates are made! feel free to reference jywarren to make final approve if you think it is ready!

**Comment by gauravano** [comment]
Hi @jywarren @publiclab/community-reps, please give a final review. Thanks!

File: doc/CHECKINS.md [hunk 3]
```diff
@@ -32,13 +32,8 @@ notify everyone by referencing `@/all` and linking the post. Include a brief mes
 
   - Troubleshot any issues that the rest of the community can learn from?
 
-- Include sections from the last post that make sense to carry over - if you need additional guidance reference past check-ins.
+- Include sections from the [last check-in](https://github.com/publiclab/plots2/issues?q=is%3Aopen+is%3Aissue+label%3A%22community+check-in%22) that make sense to carry over. If you need hint for check-in template, theme or content, refer [past check-ins](https://github.com/publiclab/plots2/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22community+check-in%22+).
```
Comment (sashadev-sky) [inline doc/CHECKINS.md]: Just a small change - 
1) if you need _a_ hint, _reference_ the template, theme, or content of [past check-ins](https://github.com/publiclab/plots2/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22community+check-in%22+)

File: doc/CHECKINS.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+- Add anything else you'd like. 
+
+  - Discovered a new technology you'd like to share? 
+
+  - Found an interesting article about open source? 
+
+  - Have any questions? 
+
+  - Troubleshot any issues that the rest of the community can learn from?
```
Comment (rexagod) [inline doc/CHECKINS.md]: A small change required here: *Troubleshoot

File: doc/CHECKINS.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+- Check-ins are typically opened as issues anytime on Sunday or Monday.
+
+- Volunteer to open the next check-in by leaving a comment on the current one.
+
+-  When you open a check-in please use our [Gitter](https://gitter.im/publiclab/publiclab) chatroom to
+notify everyone by referencing `@/all` and linking the post. Include a brief message for new contributors so that they know what it is. ex:
+
+"<em>Hey! Please see our weekly check-in \<here\>. Everyone is welcome to contribute and we encourage new users to get involved</em>."
+
+- Check-ins have a <em>theme</em>. Select any theme that interests you to guide the general flow of responces from the communtiy.
```
Comment (rexagod) [inline doc/CHECKINS.md]: Two edits here: *responses, *community

File: doc/CHECKINS.md (hunk 3 above)
Comment (gauravano) [reply doc/CHECKINS.md]: OK. I am changing this line. :+1:

File: doc/CHECKINS.md (hunk 4 above)
Comment (gauravano) [reply doc/CHECKINS.md]: It's not a typo.

File: doc/CHECKINS.md (hunk 5 above)
Comment (gauravano) [reply doc/CHECKINS.md]: Nice catch! Correcting both.

File: doc/CHECKINS.md (hunk 3 above)
Comment (gauravano) [reply doc/CHECKINS.md]: Done.

File: doc/CHECKINS.md (hunk 4 above)
Comment (rexagod) [reply doc/CHECKINS.md]: Ah I see, it's correct indeed! Apologies @gauravano, I didn't read that sentence completely. :+1:

**Comment by jywarren** [comment]
Sorry for being a bit slow to reply here! Re: `guide.publiclab.org` idea - very cool and i appreciate your interest in not fragmenting across multiple platforms. Here's a quick suggestion although we might also want to open a new issue... very good idea... what about using GitHub project boards, where we might be able to present a kind of dynamic menu of support topics, while still staying in the usual GitHub issues interface: https://github.com/publiclab/plots2/projects/4

As for this PR, i'll do a quick review but overall, tremendous work, and MUCH appreciated!!!!

**Comment by jywarren** [comment]
🎉 👍 👍 👍 👍 👍

**Comment by sashadev-sky** [comment]
@jywarren I like the github boards a lot! I have looked into them before. They are limited in terms of what you can do and customizability so we would have to go outside of the box for many of the ideas contributors have mentioned, but I think they'd be great to have and a nice interface! 

There is already a [plots2 project board](https://github.com/publiclab/plots2/projects/1?fullscreen=true). It might be jywarrens personal organizing tool, but if people like it we can make another one tailored for community interaction (you can write notes on it which everyone can see which is cool)

@rexagod

**Pull request merged**